internal/services/cognitive/cognitive_deployment_resource.go

```diff
@@ -94,21 +99,46 @@ func (r CognitiveDeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 			},
 		},
 
-		"scale": {
+		"sku": {
```

Comment:
Completely removing/renaming will break existing users since this is ForceNew - since these look to be doing/controlling the same thing I think we should keep the existing names but map it to the new names e.g. scale -> sku and type -> name within the provider, this means users configs will remain the same. We can do a hard rename of this in the next major release. 

Does that makes sense?

Reply:
It appears we're also blocked by this and can't deploy changes to our OpenAI deployments via Terraform.

Is there a short-term workaround to make 'scale' not required, potentially?

Reply:
@stephybun I've tried the solution you suggested but we have a problem now.  There's an argument named `capacity` in the new `sku` block:

```hcl
"capacity": {
				Type:     pluginsdk.TypeInt,
				Optional: true,
				ForceNew: true,
				Default:  1,
			},
```

The `capacity`'s default value is `1`, and you cannot pass `nil` to API otherwise an error would be raised. That means the whole new `sku` block is required, otherwise when we try to import the resource and got `1` for `capacity` from the service side, the absent of `sku` block would cause a drift. Since all properties for this resource are ForceNew that would cause a re-creation.

But if users added a `sku` block to the existing code, that would cause a re-creation too. It seems like we cannot avoid a re-creation.

The good news is this deployment resource seems to be a stateless one, so a re-creation won't cause any data loss or long-term service downtime.

Another possible approach is just to hide all `sku`'s properties, and just hard code the `capacity` to `1` in the provider's code, that looks better than the status quo to me.

I'll try this approach and you can decide whether you like this design or not.

Reply:
That sounds like a reasonable plan to me

Reply:
@katbyte , Is there any process to track these kinds of improvements/breaking change to 4.0 ?